### ISSUE TITLE
How can I get a cutout from compositing alpha image layer with original

### ISSUE THREAD

**Issue by whimzyLive** [lovell/sharp#2403]
# tl;dr
I need to composite two images that would give me a cutout image.

## More info
I have two images that I want to blend in to get a cutout of matching sections. One is original image and another is, area of original image to cut out in alpha layer. 

something like below.

### Original Image
![color](https://user-images.githubusercontent.com/22066605/95280736-4080c280-089d-11eb-9d24-f0d55fbc6482.jpg)

### Only Alpha layer 
![alpha](https://user-images.githubusercontent.com/22066605/95280775-5098a200-089d-11eb-9988-440782ccd6ce.png)

### Expected output
![no-bg](https://user-images.githubusercontent.com/22066605/95284665-4202b880-08a6-11eb-934b-ee4834c0aeca.png)

With imagemagick I could do something like this,

```bash
  convert color.jpg alpha.png             \
    -compose CopyOpacity                  \
    -composite no-bg.png
```

How would this work with sharp?

I have tried this so far

```typescript
    await originalImage
      .composite([
        {
          input: alphaImageBuffer,
          blend: 'dest-in'
        }
      ])
      .toFile(`${filesDirPath}/no-bg.png`);
```

Am I doing something wrong here?

**Comment by lovell** [comment]
Yes, this is the correct approach. The `alphaImageBuffer` image requires an alpha channel, but the "Only Alpha layer" example provided in this issue does not contain one (it's a single channel, greyscale image).

**Comment by whimzyLive** [comment]
I ended up doing this: (*due to my limited understanding of how images work with layers, and with some help from my friend)

Just posting an answer here, as it might be useful to someone else :)

Final code
```ts

const alphaImage = sharp(`${filesDirPath}/alpha.png`);

//  make black pixels transparent and change RGBs to white 
    const { data, info } = await alphaImage
      .ensureAlpha()
      .raw()
      .toBuffer({
        resolveWithObject: true
      });
    for (let i = 0; i < data.length; i = i + 4) {
      data[i + 3] = data[i + 0]; // move to alpha
      data[i + 0] = data[i + 1] = data[i + 2] = 255; // turns RGB into white
    }
    await sharp(data, {
      raw: {
        ...(info as any) // raw.channel and info.channel has different types (1 | 2 |3 |4 vs number)
      }
    })
      .png()
      .toFile(`${filesDirPath}/transparent.png`);


// composite transparent layer with original image
const colorImage = sharp(`${filesDirPath}/color.jpg`);
  await colorImage
      .composite([
        {
          input: `${filesDirPath}/transparent.png`,
          blend: 'dest-in'
        }
      ])
      .toFile(`${filesDirPath}/no-bg.png`);
```
@lovell  Thanks.

**Comment by whimzyLive** [comment]
@lovell also forgot to mention but `channels` prop in `raw` expects type `1 | 2 | 3 | 4` and returned `channels` prop from `toBuffer` has type `number`, thus ` ...(info as any)`. Will do a PR later this week with a fix.
